### PR TITLE
Add instruction on sending push notification when message is saved

### PR DESCRIPTION
**Pull request by b123400**
connect https://github.com/SkygearIO/chat/issues/42

The code:
- Send notifications in the following formats:
  - `<author> :<message>`
  - `<author>@<conversation_name> :<message>`
- Set badge number to total unread message count

### CODE REVIEW

File: chat/user_conversation.py [hunk 1]
```diff
@@ -58,8 +58,7 @@ def delete(self):
         })
 
 
-def total_unread():
-    user_id = current_user_id()
+def total_unread(user_id=current_user_id()):
```
Comment (cheungpat) [inline chat/user_conversation.py]: the default value of the `user_id` argument is evaluated when the function is declared, but the default value of the `current_user_id()` should depends one the context

File: README.md [hunk 2]
```diff
@@ -62,6 +62,71 @@ def chat_demo():
 
 `https://<your_app_name>.skygeario.com/static/demo/index.html`
 
+### Sending push notification
+
+Push notification can be implemented by the following code.
+Please ensure APNS certificate and private key are properly setup,
+if you are using [Skygear.io], you can configure it at the setting panel.
```
Comment (cheungpat) [inline README.md]: in the Settings panel. (and I am not sure if that is called a panel, but I could go with that)

File: README.md [hunk 3]
```diff
@@ -62,6 +62,71 @@ def chat_demo():
...
+
+```python
+from skygear.container import SkygearContainer
+from skygear.options import options as skyoptions
+from skygear.action import push_user
+
+# Please modify these import statment according to your project structure
```
Comment (cheungpat) [inline README.md]: You can assume that skygear-cms is installed as a package (using pip/setup.py) so the package name will be `chat.*`.

You can also make the comment better by saying in what scenario should developer use relative imports (“If you installed the plugin using git submodule, add `.` to use relative import.”)

File: chat/user_conversation.py (hunk 1 above)
Comment (b123400) [inline chat/user_conversation.py]: Wow I didn't know about this!

**Comment by b123400** [comment]
Updated~

I did not mention about git submodule, because submodule can be placed inside a subdirectory, and the point of the import statement is not really about git, but the relative path.

**Pull request merged**